File: android/app/src/main/java/jaclib/memory/Stream.java
package jaclib.memory;


public final class Stream {

    public Buffer a;

    public int b;

    public int e;

    public int d;

    public final byte[] c;

    public static byte getLSB(int arg0) { return 0; }

    public static int floatToRawIntBits(float arg0) { return 0; }

    public int b() {
        return this.b + this.d;
    }

    public Stream(Buffer arg0) {
        this(arg0, 0, arg0.getSize());
    }

    public Stream(Buffer arg0, int arg1, int arg2) {
        this(arg0.getSize() >= 4096 ? 4096 : arg0.getSize());
        this.a(arg0, arg1, arg2);
    }

    public Stream() {
        this(4096);
    }

    public Stream(int arg0) {
        this.c = new byte[arg0];
    }

    public void a(float arg0) {
        if (this.b + 3 >= this.c.length) {
            this.flush();
        }
        int local14 = floatToRawIntBits(arg0);
        this.c[this.b++] = (byte) (local14 >> 24);
        this.c[this.b++] = (byte) (local14 >> 16);
        this.c[this.b++] = (byte) (local14 >> 8);
        this.c[this.b++] = (byte) local14;
    }

    public int a() {
        return this.b + this.d;
    }

    public void f(int arg0) {
        this.flush();
        this.d = arg0;
    }

    public void b(int arg0) {
        if (this.c.length <= this.b + 1) {
            this.flush();
        }
        this.c[this.b++] = (byte) (arg0 >> 8);
        this.c[this.b++] = (byte) arg0;
    }

    public void c(int arg0) {
        if (this.c.length <= this.b + 1) {
            this.flush();
        }
        this.c[this.b++] = (byte) arg0;
        this.c[this.b++] = (byte) (arg0 >> 8);
    }

    public void a(Buffer arg0) {
        this.a(arg0, 0, arg0.getSize());
    }

    public void a(int arg0, int arg1, int arg2, int arg3) {
        if (this.b + 3 >= this.c.length) {
            this.flush();
        }
        this.c[this.b++] = (byte) arg2;
        this.c[this.b++] = (byte) arg1;
        this.c[this.b++] = (byte) arg0;
        this.c[this.b++] = (byte) arg3;
    }

    public void a(int arg0) {
        if (this.c.length <= this.b + 3) {
            this.flush();
        }
        this.c[this.b++] = (byte) arg0;
        this.c[this.b++] = (byte) (arg0 >> 8);
        this.c[this.b++] = (byte) (arg0 >> 16);
        this.c[this.b++] = (byte) (arg0 >> 24);
    }

    public void a(Buffer arg0, int arg1, int arg2) {
        this.flush();
        this.a = arg0;
        this.e = arg2 + arg1;
        this.d = arg1;
        if (this.e > arg0.getSize()) {
            throw new RuntimeException();
        }
    }

    public void e(int arg0) {
        if (this.b + 3 >= this.c.length) {
            this.flush();
        }
        this.c[this.b++] = (byte) (arg0 >> 16);
        this.c[this.b++] = (byte) (arg0 >> 8);
        this.c[this.b++] = (byte) arg0;
        this.c[this.b++] = (byte) (arg0 >> 24);
    }

    public void d(int arg0) {
        if (this.b >= this.c.length) {
            this.flush();
        }
        this.c[this.b++] = (byte) arg0;
    }

    public void b(int arg0, int arg1, int arg2, int arg3) {
        if (this.c.length <= this.b + 3) {
            this.flush();
        }
        this.c[this.b++] = (byte) arg0;
        this.c[this.b++] = (byte) arg1;
        this.c[this.b++] = (byte) arg2;
        this.c[this.b++] = (byte) arg3;
    }

    public void b(float arg0) {
        if (this.b + 3 >= this.c.length) {
            this.flush();
        }
        int local18 = floatToRawIntBits(arg0);
        this.c[this.b++] = (byte) local18;
        this.c[this.b++] = (byte) (local18 >> 8);
        this.c[this.b++] = (byte) (local18 >> 16);
        this.c[this.b++] = (byte) (local18 >> 24);
    }

    public void flush() {
        if (this.b <= 0) {
            return;
        }
        if (this.d + this.b > this.e) {
            throw new RuntimeException();
        }
        this.a.a(this.c, 0, this.d, this.b);
        this.d += this.b;
        this.b = 0;
    }
    public static boolean c() { return true; }
}

File: src/jvmMain/kotlin/jaclib/memory/Stream.kt
package jaclib.memory

import kotlin.jvm.JvmStatic

actual class Stream /*private*/ actual constructor(arg0: Int) {
    /*private*/ actual var a: Int = 0

    /*private*/ actual var b: Buffer? = null

    /*private*/ actual var c: Int = 0

    /*private*/ actual var d: Int = 0

    /*private*/ actual val e: ByteArray

    actual constructor() : this(4096)

    @JvmOverloads
    actual constructor(arg0: Buffer, arg1: Int, arg2: Int) : this(if (arg0.getSize() >= 4096) 4096 else arg0.getSize()) {
        this.a(arg0, arg1, arg2)
    }

    init {
        this.e = ByteArray(arg0)
    }

    actual fun a(arg0: Int) {
        if (this.c + 1 >= this.e.size) {
            this.a()
        }
        this.e[this.c++] = arg0.toByte()
        this.e[this.c++] = (arg0 shr 8).toByte()
    }

    actual fun a(arg0: Float) {
        if (this.e.size <= this.c + 3) {
            this.a()
        }
        val local12: Int = floatToRawIntBits(arg0)
        this.e[this.c++] = (local12 shr 24).toByte()
        this.e[this.c++] = (local12 shr 16).toByte()
        this.e[this.c++] = (local12 shr 8).toByte()
        this.e[this.c++] = local12.toByte()
    }

    actual fun a(arg0: Int, arg1: Int, arg2: Int, arg3: Int) {
        if (this.c + 3 >= this.e.size) {
            this.a()
        }
        this.e[this.c++] = arg0.toByte()
        this.e[this.c++] = arg1.toByte()
        this.e[this.c++] = arg2.toByte()
        this.e[this.c++] = arg3.toByte()
    }

    actual fun a(arg0: Buffer) {
        this.a(arg0, 0, arg0.getSize())
    }

    actual fun b(arg0: Int) {
        if (this.c + 3 >= this.e.size) {
            this.a()
        }
        this.e[this.c++] = arg0.toByte()
        this.e[this.c++] = (arg0 shr 8).toByte()
        this.e[this.c++] = (arg0 shr 16).toByte()
        this.e[this.c++] = (arg0 shr 24).toByte()
    }

    actual fun a() {
        if (this.c <= 0) {
            return
        }
        if (this.a < this.c + this.d) {
            throw RuntimeException()
        }
        this.b!!.a(this.e, 0, this.d, this.c)
        this.d += this.c
        this.c = 0
    }

    /*private*/ actual fun a(arg0: Buffer, arg1: Int, arg2: Int) {
        this.a()
        this.a = arg1 + arg2
        this.d = arg1
        this.b = arg0
        if (this.a > arg0.getSize()) {
            throw RuntimeException()
        }
    }

    actual fun b(arg0: Int, arg1: Int, arg2: Int, arg3: Int) {
        if (this.e.size <= this.c + 3) {
            this.a()
        }
        this.e[this.c++] = arg2.toByte()
        this.e[this.c++] = arg1.toByte()
        this.e[this.c++] = arg0.toByte()
        this.e[this.c++] = arg3.toByte()
    }

    actual fun c(arg0: Int) {
        if (this.e.size <= this.c + 3) {
            this.a()
        }
        this.e[this.c++] = (arg0 shr 16).toByte()
        this.e[this.c++] = (arg0 shr 8).toByte()
        this.e[this.c++] = arg0.toByte()
        this.e[this.c++] = (arg0 shr 24).toByte()
    }

    actual fun b(): Int {
        return this.d + this.c
    }

    actual fun d(arg0: Int) {
        if (this.e.size <= this.c + 1) {
            this.a()
        }
        this.e[this.c++] = (arg0 shr 8).toByte()
        this.e[this.c++] = arg0.toByte()
    }

    actual fun e(arg0: Int) {
        this.a()
        this.d = arg0
    }

    actual fun b(arg0: Float) {
        if (this.e.size <= this.c + 3) {
            this.a()
        }
        val local12: Int = floatToRawIntBits(arg0)
        this.e[this.c++] = local12.toByte()
        this.e[this.c++] = (local12 shr 8).toByte()
        this.e[this.c++] = (local12 shr 16).toByte()
        this.e[this.c++] = (local12 shr 24).toByte()
    }

    actual fun f(arg0: Int) {
        if (this.c >= this.e.size) {
            this.a()
        }
        this.e[this.c++] = arg0.toByte()
    }

    actual companion object {
        @JvmStatic
        actual external fun floatToRawIntBits(arg0: Float): Int

        @JvmStatic
        /*private*/ actual external fun getLSB(arg0: Int): Byte

        @JvmStatic
        actual fun c(): Boolean {
            return getLSB(-65536).toInt() == -1
        }
    }
}
